### PR TITLE
Bug 1800611 - Let Android-Components use gecko version numbers

### PR DESCRIPTION
**Pull request by JohanLorenzo**
Tested in staging:
 * nightly https://firefox-ci-tc.services.mozilla.com/tasks/groups/QN3uLZZLTdSzoeTVKzr2WA (the build error is unrelated, I forced the rest of the graph to run)
 * beta https://firefox-ci-tc.services.mozilla.com/tasks/groups/BmqPUzqGQuGS2RQGyOub1g

I recommend A-C 109 (current nightly) to be the first version to follow the Gecko scheme (e.g. `109.0b1`). We shipped many 108s (current beta) currently numbered 108.0.X, going back to `108.0bX` is going to be a mess to deal with. We can discuss the details in tomorrow's Android monorepo meeting.

Depends on https://github.com/mozilla-releng/mozilla-version/pull/100 and https://github.com/mozilla-mobile/relbot/pull/97

### CODE REVIEW

File: taskcluster/ac_taskgraph/loader/build_config.py [hunk 1]
```diff
@@ -120,7 +120,14 @@ def loader(kind, path, config, params, loaded_tasks):
     # Disable the affected_components optimization to make sure we execute all tests to get
     # a complete code coverage report for pushes to 'main'.
     # See https://github.com/mozilla-mobile/android-components/issues/9382#issuecomment-760506327
-    if params["tasks_for"] in ("github-pull-request", "github-pull-request-untrusted", "github-push") and params["head_ref"] not in ("main", "refs/heads/main"):
+    should_take_out_tasks = all((
+        params["tasks_for"] in ("github-pull-request", "github-pull-request-untrusted", "github-push"),
+        params["head_ref"] not in ("main", "refs/heads/main"),
+        # try_task_config.json can override `target_tasks_method`` to make staging nightly
+        # for instance
+        params["target_tasks_method"] == "default",
```
Comment (gbrownmozilla) [inline taskcluster/ac_taskgraph/loader/build_config.py:128]: Is this a separate issue / ride-along fix, or is it somehow related to the version number change? I don't object -- just checking my understanding.

**Review by gbrownmozilla — APPROVED** [review]

**Review by rvandermeulen — APPROVED** [review]
LGTM noting that my review here is more of a general approval of the plan. I agree that starting this with 109 makes the most sense.

File: taskcluster/ac_taskgraph/loader/build_config.py (hunk 1 above)
Comment (JohanLorenzo) [reply taskcluster/ac_taskgraph/loader/build_config.py:128]: Thanks for pointing this out! It's a ride-along fix. I wanted to use `try_task_config.json` to make a nightly on this PR but I stumbled upon this bug. That's why I fixed it here.

**Comment by JohanLorenzo** [comment]
We all agreed on the numbering plan during the meeting. Merging this PR.

**Pull request merged**